### PR TITLE
Support Relabel of Prometheus Metrics Tags based on provided config.

### PR DESCRIPTION
**Pull request by neerajmangal**
Add support for Metrics Tag relabeling/rename using config.  

## Description

With this PR, relabel/rename of Prometheus metrics tags can be done using the below configuration. 
Tags defined under `rename-tags` configs will be overridden before sending them to Prometheus. 

```
whisk {
  user-events {
    rename-tags {
      # rename/relabel prometheus metrics tags
      "namespace" = "ow_namespae"
    }
  }
}
``` 

For example with the above configuration, `namespace` tag will be changed to `ow_namespace` like below. 

Screenshot 2020-04-07 at 10 16 21 PM

TODO 
- [x] Required Tests.
- [x] Documentation Updates

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation
- [x] Monitoring/User-Events

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [x] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [x] I updated the documentation where necessary.

### CODE REVIEW

File: core/monitoring/user-events/src/test/resources/application.conf [hunk 1]
```diff
@@ -26,4 +26,7 @@ user-events {
   # Namespaces that should not be monitored
   ignored-namespaces = ["guest"]
 
+  rename-tags {
+    namespace = "ow_namespace"
```
Comment (selfxp) [inline core/monitoring/user-events/src/test/resources/application.conf:30]: Could we add a test that validates the rename of this tag?

File: core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/PrometheusRecorder.scala [hunk 2]
```diff
@@ -150,15 +152,111 @@ case class PrometheusRecorder(kamon: PrometheusReporter)
...
+    val metricTags: LinkedHashMap[String, String] = LinkedHashMap(
+      actionNamespace -> config.renameTags.getOrElse(actionNamespace, actionNamespace),
+      initiatorNamespace -> config.renameTags.getOrElse(initiatorNamespace, initiatorNamespace),
+      actionName -> config.renameTags.getOrElse(actionName, actionName),
+      actionKind -> config.renameTags.getOrElse(actionKind, actionKind),
+      actionMemory -> config.renameTags.getOrElse(actionMemory, actionMemory),
+      actionStatus -> config.renameTags.getOrElse(actionStatus, actionStatus),
+      userDefinedStatusCode -> config.renameTags.getOrElse(userDefinedStatusCode, userDefinedStatusCode))
+
+    val namespaceActivationTags = metricTags.filterKeys(Set(actionNamespace, initiatorNamespace)).values.toSeq
```
Comment (selfxp) [inline core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/PrometheusRecorder.scala:175]: This is just a take it or leave and a personal preference but defining a map and then filtering it for extracting the desired values seems like an optimization but it requires more time to understand what is happening there. I would just create the variables and reference them one by one in the counters and histograms as before.

**Comment by rabbah** [comment]
I’m curious what motivates this?

**Comment by neerajmangal** [comment]
> I’m curious what motivates this?

Sorry @rabbah, I should have added it in the description. 

The motivation to add relabeling of metric tags/labels comes from a problem we hit recently using external Prometheus (as a service) with Openwhisk Kubernetes Deployment. 

Openwhisk populates the `namespace` label which conflicts with the `namespace` label Kubernetes also has. The external Prometheus as a service does not expose any scraping time relabel configuration so we have added this feature to avoid such conflict.

**Comment by neerajmangal** [comment]
Hi @selfxp/@rabbah, please let me know if I can go ahead and merge this PR.

**Pull request merged**